packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -944,12 +944,14 @@ abstract class FlutterCommand extends Command<void> {
   }
 
   /// Updates dart-defines based on [webRenderer].
-  void _updateDartDefines(List<String> dartDefines, String webRenderer) {
-    if (dartDefines.any((String d) => d.startsWith('FLUTTER_WEB_USE_SKIA='))) {
+  List<String> _updateDartDefines(List<String> dartDefines, String webRenderer) {
```

Comment:
Let's change `_update` to `_cleanUp` or `_validate` since this no longer updates the incoming list.

Reply:
It still may update the list depending on the circumstance so I think `update` should be fine.